torch/nn/modules/loss.py

```diff
@@ -373,77 +373,85 @@ def forward(self, input: Tensor, target: Tensor, var: Tensor) -> Tensor:
 
 
 class KLDivLoss(_Loss):
-    r"""The Kullback-Leibler divergence loss measure
+    r"""The Kullback-Leibler divergence loss.
 
-    `Kullback-Leibler divergence`_ is a useful distance measure for continuous
-    distributions and is often useful when performing direct regression over
-    the space of (discretely sampled) continuous output distributions.
+    For tensors of the same shape :math:`y_{\text{model}},\ y_{\text{obs}}`,
```

Comment:
The math looks correct to me and I appreciate the rigor you approached this with!

Only thing that concerns me is that it's inconsistent with other losses, which generally use the notation `x_n` for input and `y_n` for target. I think it's okay that it's different, but I have a couple suggestions:
* I'd make the connection between the two variables and input / target very explicit (something like `"where y_{model} is the ``input`` and y_{obs} is the target"`)
* Possibly use `y_{true}` / `y_{pred}` instead? this matches TF and I believe fits well with how ML practitioners think (in terms of predicted values vs. ground truth)

Overall I think it's much clearer what's going on :)

Reply:
@jbschlosser, suggest actual explicit reference to the P and Q arguments and the implementation being swapped wrt. to math because this is machine learning framework tradition (cf. Tensorflow)

Reply:
The obs / model came from the probabilistic take on the KL divergence, but I see that using more layman terms might very well be helpful to many.

Also, using this we get rid of parens / em-dashes :D